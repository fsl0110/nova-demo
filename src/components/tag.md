### Select a HTML-Tag of your joice

#### With type strong

```jsx
<Tag type="strong">Text in strong tags content!</Tag>
```

#### With type h1

```jsx
<Tag type="h1">Text in h1 tags content!</Tag>
```

#### With type em

```jsx
<Tag type="em">Text in em tags content!</Tag>
```
