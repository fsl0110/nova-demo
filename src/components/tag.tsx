import React, { FC, ReactNode, HTMLAttributes } from "react";

export interface Props extends HTMLAttributes<HTMLElement> {
  /** Nodes within this component **/
  children: ReactNode | ReactNode[];
  /** Type of a HTML-Tag this component will use as root tag**/
  type: keyof JSX.IntrinsicElements;
}

export const Tag: FC<Props> = ({ type, className, children, ...rest }) =>
  React.createElement(type, {
    dangerouslySetInnerHTML: { __html: children },
    className,
    ...rest
  });

export default Tag;
