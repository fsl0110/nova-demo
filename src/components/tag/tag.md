### Select a HTML-Tag of your joice

#### Tag type strong

```jsx
<Tag type="strong">Text in strong tag.</Tag>
```

#### Tag type h1

```jsx
<Tag type="h1">Text in h1 tag.</Tag>
```

#### Tag type em

```jsx
<Tag type="em">Text in em tag.</Tag>
```
