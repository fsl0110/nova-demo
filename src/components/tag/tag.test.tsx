import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Tag, Props } from "./tag";

const testCases: Array<[string, Props]> = [
  [
    "Tag h1",
    {
      type: "h1",
      children: "some text"
    }
  ],
  [
    "Tag strong",
    {
      type: "strong",
      children: "some text"
    }
  ],
  [
    "Tag em tag",
    {
      type: "em",
      children: "some text"
    }
  ]
];

describe("<Tag>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(
        toJson(shallow(<Tag {...options}>Some text</Tag>))
      ).toMatchSnapshot();
    });
  });
});
