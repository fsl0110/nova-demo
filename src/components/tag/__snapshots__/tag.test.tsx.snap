// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> Tag em tag 1`] = `
<em
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;

exports[`<Tag> Tag h1 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;

exports[`<Tag> Tag strong 1`] = `
<strong
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;
